ai/core/src/unitycatalog/ai/core/databricks.py

```diff
@@ -181,34 +150,45 @@ def wrapper(self, *args, **kwargs):
         for attempt in range(1, max_attempts + 1):
             try:
                 result = func(self, *args, **kwargs)
-                # for non-session related error in the result, we should directly return the result
-                if (
-                    isinstance(result, FunctionExecutionResult)
-                    and result.error
-                    and SESSION_EXCEPTION_MESSAGE in result.error
-                ):
-                    raise Exception(result.error)
+                if isinstance(result, FunctionExecutionResult) and result.error:
+                    if any(
+                        msg in result.error
+                        for msg in (SESSION_EXPIRED_MESSAGE, SESSION_CHANGED_MESSAGE)
+                    ):
+                        raise Exception(result.error)
```

Comment:
Could we create a SessionExpirationException and catch it in below, so we don't need to check the message twice?

Reply:
good call. Much cleaner